aws/resource_aws_rds_cluster.go

```diff
@@ -203,12 +203,58 @@ func resourceAwsRDSCluster() *schema.Resource {
 				},
 			},
 
+			"restore_in_time": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				ConflictsWith: []string{
+					"s3_import",
+					"snapshot_identifier",
+				},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"source_cluster_identifier": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validateRdsIdentifier,
+						},
+
+						"restore_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"full-copy",
+								"copy-on-write",
+							}, false),
+						},
+
+						"use_latest_restorable_time": {
+							Type:          schema.TypeBool,
+							Optional:      true,
+							ForceNew:      true,
+							ConflictsWith: []string{"restore_in_time.restore_to_time"},
+						},
+
+						"restore_to_time": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validateRdsRestorableTimeFormat,
```

Comment:
In this PR, i've re-used  this method you have here but with a more generic name since it's applicable to other resources in the Provider: https://github.com/hashicorp/terraform-provider-aws/pull/15969/files#diff-78a0a97b5a4b0942c73e64eec6c3fbee6ce1e5e23744375b290543b2259ea320

Reply:
Shall I rename the function here? I can't use the function from your PR because it is not merged yet.

And if you don't mind me asking, your PR is essentially the same as this. Why are you reviewing this one instead of continuing on that one? I am happy that you are reviewing this one and there is a chance it might go through, but I am also curious what's wrong with the other one?

Reply:
if you don't mind and agree with the renaming! once we get this PR merged, I can rebase. 

Nothing wrong with my PR (afik atm 😅 ), just needs a review from our internal team. I'm reviewing this one in the meantime so we can hopefully get the two enhancements in to close out #5286

Reply:
@anGie44 renamed to `validateUTCTimestamp`.